compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -528,7 +528,11 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 }
             }
 
-            Drop { place, target, unwind, replace: _ } => {
+            Drop { place, target, unwind, replace: _, drop, async_fut } => {
+                assert!(
+                    async_fut.is_none() && drop.is_none(),
+                    "Async Drop must be expanded or reset to sync before consteval"
```

Comment:
```suggestion
                    "Async Drop must be expanded or reset to sync in runtime MIR"
```
This is also used by Miri, not just for consteval.

Reply:
Fixed.